From: Alek Du <alek.du@intel.com>
Date: Mon, 19 Apr 2010 12:39:37 -0800
Subject: [PATCH] ehci: Only enable QH prefetch on Moorestown
Patch-mainline: when mrst code is merged?

Hardware QH prefetch feature is found to be buggy on some none-moorestown
platforms. Disable this feature on other platforms at the moment.

Signed-off-by: Alek Du <alek.du@intel.com>
---
 drivers/usb/host/ehci-hcd.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/host/ehci-hcd.c
+++ b/drivers/usb/host/ehci-hcd.c
@@ -595,7 +595,9 @@ static int ehci_init(struct usb_hcd *hcd
 	}
 	if (HCC_HW_PREFETCH(hcc_params)) {
 		ehci_dbg(ehci, "HW prefetch capable %d\n", park);
+#ifdef CONFIG_X86_MRST
 		temp |= (CMD_ASPE | CMD_PSPE);
+#endif
 	}
 
 	if (HCC_CANPARK(hcc_params)) {
